src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenSerializer.cs

```diff
@@ -156,7 +161,14 @@ public string Serialize(AntiforgeryToken token)
 
                 writer.Flush();
                 var stream = serializationContext.Stream;
-                return WebEncoders.Base64UrlEncode(_cryptoSystem.Protect(stream.ToArray()));
+                var bytes = _cryptoSystem.Protect(stream.ToArray());
+
+                var count = bytes.Length;
+                var charsRequired = WebEncoders.GetArraySizeRequiredToEncode(count);
+                var chars = serializationContext.GetChars(charsRequired);
+                var outputLength = WebEncoders.Base64UrlEncode(bytes, 0, count, chars, 0);
+
+                return new string(chars, 0, outputLength);
```

Comment:
First, no, I don't think we should work to avoid this allocation. We'd have a hideous public API if we used almost anything but `string`s.

Second, the impact is a bit under 2 MB (~1.4% of total allocations) in the test scenario:
![after strings](https://cloud.githubusercontent.com/assets/6431421/13098886/ad48149e-d4e1-11e5-9a1d-4ebf111df43b.png)